In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("texas-dm-20110101-20110721.csv")
df

MapDate StateAbbreviation   None      D0      D1     D2     D3     D4  \
0   20110719                TX   0.00  100.00   98.96  95.78  91.19  75.16   
1   20110712                TX   0.00  100.00   97.43  95.78  90.97  71.66   
2   20110705                TX   2.41   97.59   95.73  94.39  90.21  71.30   
3   20110628                TX   2.68   97.32   95.71  94.52  90.62  72.32   
4   20110621                TX   3.33   96.67   95.71  94.52  91.31  70.61   
5   20110614                TX   1.98   98.02   96.52  94.76  88.56  64.78   
6   20110607                TX   1.98   98.02   96.52  94.04  85.41  57.83   
7   20110531                TX   2.25   97.75   96.07  91.89  81.09  50.65   
8   20110524                TX   2.30   97.70   95.77  91.97  78.48  43.97   
9   20110517                TX   0.00  100.00   97.01  92.40  80.01  47.87   
10  20110510                TX   0.00  100.00   97.78  93.88  82.06  47.55   
11  20110503                TX   0.00  100.00   98.86  93.99  73.74  25.96   
12  20110426                TX   0.00  100.00  100.00  94.97  70.42  17.16   
13  20110419                TX   0.00  100.00  100.00  91.52  67.97  14.99   
14  20110412                TX   0.00  100.00   97.94  86.44  60.58  10.03   
15  20110405                TX   0.00  100.00   97.99  86.50  60.15   4.82   
16  20110329                TX   0.00  100.00   94.87  78.54  43.07   0.00   
17  20110322                TX   1.70   98.30   92.05  64.06  28.98   0.00   
18  20110315                TX   3.23   96.77   83.75  56.99  17.13   0.00   
19  20110308                TX   8.31   91.69   77.99  53.64  14.16   0.00   
20  20110301                TX   4.99   95.01   72.91  46.35  12.72   0.00   
21  20110222                TX   5.08   94.92   69.75  45.25  12.64   0.00   
22  20110215                TX  12.52   87.48   58.12  37.35   7.78   0.00   
23  20110208                TX  15.72   84.28   57.15  37.35   7.78   0.00   
24  20110201                TX  19.47   80.53   53.00  35.11   6.93   0.00   
25  20110125                TX  19.47   80.53   53.00  31.56   8.66   0.00   
26  20110118                TX  23.28   76.72   51.42  25.62   7.30   0.00   
27  20110111                TX  20.16   79.84   59.13  25.37   9.48   0.00   
28  20110104                TX  13.54   86.46   66.68  36.30  13.04   0.00   
29  20101228                TX   7.89   92.11   69.44  37.46   9.59   0.00   

    ValidStart    ValidEnd  StatisticFormatID  
0   2011-07-19  2011-07-25                  1  
1   2011-07-12  2011-07-18                  1  
2   2011-07-05  2011-07-11                  1  
3   2011-06-28  2011-07-04                  1  
4   2011-06-21  2011-06-27                  1  
5   2011-06-14  2011-06-20                  1  
6   2011-06-07  2011-06-13                  1  
7   2011-05-31  2011-06-06                  1  
8   2011-05-24  2011-05-30                  1  
9   2011-05-17  2011-05-23                  1  
10  2011-05-10  2011-05-16                  1  
11  2011-05-03  2011-05-09                  1  
12  2011-04-26  2011-05-02                  1  
13  2011-04-19  2011-04-25                  1  
14  2011-04-12  2011-04-18                  1  
15  2011-04-05  2011-04-11                  1  
16  2011-03-29  2011-04-04                  1  
17  2011-03-22  2011-03-28                  1  
18  2011-03-15  2011-03-21                  1  
19  2011-03-08  2011-03-14                  1  
20  2011-03-01  2011-03-07                  1  
21  2011-02-22  2011-02-28                  1  
22  2011-02-15  2011-02-21                  1  
23  2011-02-08  2011-02-14                  1  
24  2011-02-01  2011-02-07                  1  
25  2011-01-25  2011-01-31                  1  
26  2011-01-18  2011-01-24                  1  
27  2011-01-11  2011-01-17                  1  
28  2011-01-04  2011-01-10                  1  
29  2010-12-28  2011-01-03                  1

In [96]:
df = df.drop(['StatisticFormatID'], axis=1)

In [4]:
df['month'] = pd.to_datetime(df.MapDate, format='%Y%m%d').dt.strftime('%m')
df.sample(5)

MapDate StateAbbreviation   None      D0     D1     D2     D3     D4  \
15  20110405                TX   0.00  100.00  97.99  86.50  60.15   4.82   
4   20110621                TX   3.33   96.67  95.71  94.52  91.31  70.61   
26  20110118                TX  23.28   76.72  51.42  25.62   7.30   0.00   
0   20110719                TX   0.00  100.00  98.96  95.78  91.19  75.16   
18  20110315                TX   3.23   96.77  83.75  56.99  17.13   0.00   

    ValidStart    ValidEnd  StatisticFormatID month  
15  2011-04-05  2011-04-11                  1    04  
4   2011-06-21  2011-06-27                  1    06  
26  2011-01-18  2011-01-24                  1    01  
0   2011-07-19  2011-07-25                  1    07  
18  2011-03-15  2011-03-21                  1    03

In [5]:
df2 = pd.read_csv("texas-dm-20220101-20220721.csv")
df2

MapDate StateAbbreviation   None     D0     D1     D2     D3     D4  \
0   20220719                TX   0.82  99.18  94.77  84.08  56.67  20.40   
1   20220712                TX   0.76  99.24  93.82  75.70  51.80  21.32   
2   20220705                TX   2.47  97.53  86.79  66.05  45.91  16.11   
3   20220628                TX   3.71  96.29  86.39  64.99  43.79  15.82   
4   20220621                TX   7.01  92.99  81.18  64.62  43.83  17.11   
5   20220614                TX   8.59  91.41  79.97  64.03  42.47  16.78   
6   20220607                TX  11.75  88.25  78.81  64.99  40.11  15.60   
7   20220531                TX  14.11  85.89  78.44  66.35  44.07  17.91   
8   20220524                TX  12.92  87.08  79.11  66.94  46.05  22.08   
9   20220517                TX   9.57  90.43  81.42  69.47  55.79  28.91   
10  20220510                TX  10.54  89.46  79.23  68.09  52.96  24.53   
11  20220503                TX   8.83  91.17  80.02  67.29  50.91  23.19   
12  20220426                TX   6.21  93.79  82.11  70.28  50.25  19.62   
13  20220419                TX   3.87  96.13  87.38  72.46  53.54  15.91   
14  20220412                TX   2.87  97.13  87.66  74.12  49.11  14.20   
15  20220405                TX   4.95  95.05  84.73  71.45  40.56   9.78   
16  20220329                TX   4.90  95.10  88.22  70.79  42.10   7.03   
17  20220322                TX   4.28  95.72  88.24  71.31  40.38   6.20   
18  20220315                TX   3.69  96.31  91.22  70.13  41.20   8.63   
19  20220308                TX   3.95  96.05  89.93  68.43  36.38   6.39   
20  20220301                TX   6.66  93.34  80.71  56.71  24.47   0.00   
21  20220222                TX   7.67  92.33  79.06  52.89  22.44   0.00   
22  20220215                TX  11.83  88.17  77.61  55.58  24.53   0.00   
23  20220208                TX  11.83  88.17  78.09  55.02  23.88   0.00   
24  20220201                TX   7.04  92.96  83.79  69.20  31.56   0.00   
25  20220125                TX   3.71  96.29  86.71  67.46  31.84   0.00   
26  20220118                TX   3.53  96.47  86.04  64.67  25.87   0.00   
27  20220111                TX   3.21  96.79  82.48  62.44  21.91   0.00   
28  20220104                TX   7.58  92.42  79.83  54.25  16.69   0.00   
29  20211228                TX  13.02  86.98  67.27  36.58  10.65   0.00   

    ValidStart    ValidEnd  StatisticFormatID  
0   2022-07-19  2022-07-25                  1  
1   2022-07-12  2022-07-18                  1  
2   2022-07-05  2022-07-11                  1  
3   2022-06-28  2022-07-04                  1  
4   2022-06-21  2022-06-27                  1  
5   2022-06-14  2022-06-20                  1  
6   2022-06-07  2022-06-13                  1  
7   2022-05-31  2022-06-06                  1  
8   2022-05-24  2022-05-30                  1  
9   2022-05-17  2022-05-23                  1  
10  2022-05-10  2022-05-16                  1  
11  2022-05-03  2022-05-09                  1  
12  2022-04-26  2022-05-02                  1  
13  2022-04-19  2022-04-25                  1  
14  2022-04-12  2022-04-18                  1  
15  2022-04-05  2022-04-11                  1  
16  2022-03-29  2022-04-04                  1  
17  2022-03-22  2022-03-28                  1  
18  2022-03-15  2022-03-21                  1  
19  2022-03-08  2022-03-14                  1  
20  2022-03-01  2022-03-07                  1  
21  2022-02-22  2022-02-28                  1  
22  2022-02-15  2022-02-21                  1  
23  2022-02-08  2022-02-14                  1  
24  2022-02-01  2022-02-07                  1  
25  2022-01-25  2022-01-31                  1  
26  2022-01-18  2022-01-24                  1  
27  2022-01-11  2022-01-17                  1  
28  2022-01-04  2022-01-10                  1  
29  2021-12-28  2022-01-03                  1

In [6]:
df2 = df2.drop(['StatisticFormatID'], axis=1)

In [7]:
df2['month'] = pd.to_datetime(df2.MapDate, format='%Y%m%d').dt.strftime('%m').copy(deep=True)
df2.sample(5)

MapDate StateAbbreviation   None     D0     D1     D2     D3     D4  \
10  20220510                TX  10.54  89.46  79.23  68.09  52.96  24.53   
20  20220301                TX   6.66  93.34  80.71  56.71  24.47   0.00   
12  20220426                TX   6.21  93.79  82.11  70.28  50.25  19.62   
3   20220628                TX   3.71  96.29  86.39  64.99  43.79  15.82   
25  20220125                TX   3.71  96.29  86.71  67.46  31.84   0.00   

    ValidStart    ValidEnd month  
10  2022-05-10  2022-05-16    05  
20  2022-03-01  2022-03-07    03  
12  2022-04-26  2022-05-02    04  
3   2022-06-28  2022-07-04    06  
25  2022-01-25  2022-01-31    01

In [153]:
df.groupby('month').D0.mean().sort_values(ascending=False)

month
April       100.000000
July         99.196667
May          99.090000
June         97.507500
March        96.354000
December     92.110000
February     86.802500
January      80.887500
Name: D0, dtype: float64

In [102]:
df.groupby('month').D0.median().sort_values(ascending=False)

month
April       100.000
July        100.000
May         100.000
June         97.670
March        96.770
December     92.110
February     85.880
January      80.185
Name: D0, dtype: float64

In [103]:
df.groupby('month').D4.mean().sort_values(ascending=False)

month
July        72.706667
June        66.385000
May         43.200000
April       11.750000
December     0.000000
February     0.000000
January      0.000000
March        0.000000
Name: D4, dtype: float64

In [104]:
df.groupby('month').D4.median().sort_values(ascending=False)

month
July        71.660
June        67.695
May         47.550
April       12.510
December     0.000
February     0.000
January      0.000
March        0.000
Name: D4, dtype: float64

# Calculate rate of growth
* To calculate growth rate: new value - old value / old value
* For D0 rate, starting from January, when we have a full month of values
* For D4 rate, starting from April, when a value first appeared in 2011

In [105]:
(df.query("month == 'July'").D0.mean()-df.query("month == 'January'").D0.mean())/df.query("month == 'January'").D0.mean()

0.2263534744758667

In [106]:
(df.query("month == 'July'").D4.mean()-df.query("month == 'April'").D4.mean())/df.query("month == 'April'").D4.mean()

5.187801418439716

In [107]:
# Adding an extra calculation between April and May where I see the biggest jump in D4 mean values
(df.query("month == 'May'").D4.mean()-df.query("month == 'April'").D4.mean())/df.query("month == 'April'").D4.mean()

2.6765957446808515

In [108]:
# Actually going to do it month to month...
(df.query("month == 'June'").D4.mean()-df.query("month == 'May'").D4.mean())/df.query("month == 'May'").D4.mean()

0.5366898148148148

In [109]:
(df.query("month == 'July'").D4.mean()-df.query("month == 'June'").D4.mean())/df.query("month == 'June'").D4.mean()

0.09522733549245549

# Changed original df datetime
To create new dataframe for viz purposes (with help from TA Jessie Blaeser), we changed the original
block of code df2['month'] = pd.to_datetime(df2.MapDate, format='%Y%m%d').dt.strftime('%B') to
df2['month'] = pd.to_datetime(df2.MapDate, format='%Y%m%d').dt.strftime('%m').copy(deep=True)
This allowed the months to be manipulated much more easily.

Also below we go through the work to create one line of code for rate of growth, change with .pct_change()


In [160]:
# Creating one line of code for rate of growth
# .to_frame() to show what data looks like in a dataframe
df22 = df2.groupby('month').D4.mean().to_frame().reset_index()

In [161]:
df22.dtypes

month     object
D4       float64
dtype: object

In [168]:
df22 = df22.sort_values('month')
df22

month         D4
0    01   0.000000
1    02   0.000000
2    03   5.650000
3    04  14.877500
4    05  23.324000
5    06  16.327500
6    07  19.276667
7    12   0.000000

In [170]:
df22 = df22[(df22['month'] > '02') & (df22['month'] <'08')]

In [171]:
df22

month         D4
2    03   5.650000
3    04  14.877500
4    05  23.324000
5    06  16.327500
6    07  19.276667

In [184]:
# when using .to_frame() to create a new df with given data, it will make the first column the index,
# we use reset_index() to make a new index and keep the columns, tho the column is renamed index
df22.month
df22 = df22.D4.pct_change().to_frame().reset_index()

# Texas 2022 drought observations

* Rate of growth for the state under any drought category (represented by D0) was 3.3% in a 7-month period
* Rate of growth for the state under exceptional drought category (D4) was 29.6% in a 4-month period or 241.2% in a 5-month period beginning in March when D4 category first began in 2022.
* Biggest jump was March to April with 163.3%
* Then it was 56.8% April to May, -30% to June, and 18.1% to July

In [183]:
# Creating one line of code for rate of growth
# .to_frame() to show what data looks like in a dataframe
df11 = df.groupby('month').D4.mean().to_frame().reset_index()
df11 = df11.sort_values('month')
df11 = df11[(df11['month'] > '02') & (df11['month'] <'08')]
df11 = df11.D4.pct_change().to_frame().reset_index()

In [185]:
df11.rename(columns={'index':'month', 'D4': 'D411_pctchng'}, inplace=True)
df22.rename(columns={'index':'month', 'D4': 'D422_pctchng'}, inplace=True)
merged = df11.merge(df22, on='month')
merged.head()

month  D411_pctchng  D422_pctchng
0      2           NaN           NaN
1      3           inf      1.633186
2      4      2.676596      0.567737
3      5      0.536690     -0.299970
4      6      0.095227      0.180626

In [186]:
merged.to_csv('texas-D4-rate-change.csv')

# Texas 2011 drought observations

* Rate of growth for the state under any drought category (represented by D0) was 22.7% in a 7-month period
* Rate of growth for the state under exceptional drought category (D4) was 518.8% in a 4-month period
* Biggest jump in D4 was April to May with 267.7%
* Then it was 53.7% to June followed by 9.5% to July
    * Basically it slowed down a lot

## Maybe I should do D0 in the 4-month period???
... fine I'll do it

In [110]:
(df.query("month == 'July'").D0.mean()-df.query("month == 'April'").D0.mean())/df.query("month == 'April'").D0.mean()

-0.008033333333333274

... wild

## Let's move on to 2022...

In [111]:
df2.groupby('month').D0.mean().sort_values(ascending=False)

month
July        98.6500
April       95.5250
January     95.4925
March       95.3040
June        92.2350
February    90.4075
May         88.8060
December    86.9800
Name: D0, dtype: float64

In [112]:
df2.groupby('month').D0.median().sort_values(ascending=False)

month
July        99.18
January     96.38
March       95.72
April       95.59
June        92.20
February    90.25
May         89.46
December    86.98
Name: D0, dtype: float64

In [113]:
df2.groupby('month').D4.mean().sort_values(ascending=False)

month
May         23.324000
July        19.276667
June        16.327500
April       14.877500
March        5.650000
December     0.000000
February     0.000000
January      0.000000
Name: D4, dtype: float64

In [114]:
df2.groupby('month').D4.median().sort_values(ascending=False)

month
May         23.190
July        20.400
June        16.300
April       15.055
March        6.390
December     0.000
February     0.000
January      0.000
Name: D4, dtype: float64

In [115]:
(df2.query("month == 'July'").D0.mean()-df2.query("month == 'January'").D0.mean())/df2.query("month == 'January'").D0.mean()

0.03306542398617706

In [116]:
(df2.query("month == 'July'").D4.mean()-df2.query("month == 'April'").D4.mean())/df2.query("month == 'April'").D4.mean()

0.29569260068335845

In [117]:
#Actually doing another calculation for 2022 because the D4 drought began in March, not April...
(df2.query("month == 'July'").D4.mean()-df2.query("month == 'March'").D4.mean())/df2.query("month == 'March'").D4.mean()

2.4117994100294986

In [118]:
# Going to add month to month like I did for 2011
(df2.query("month == 'April'").D4.mean()-df2.query("month == 'March'").D4.mean())/df2.query("month == 'March'").D4.mean()

1.6331858407079647

In [119]:
(df2.query("month == 'May'").D4.mean()-df2.query("month == 'April'").D4.mean())/df2.query("month == 'April'").D4.mean()

0.5677365148714499

In [120]:
(df2.query("month == 'June'").D4.mean()-df2.query("month == 'May'").D4.mean())/df2.query("month == 'May'").D4.mean()

-0.299969987995198

In [121]:
(df2.query("month == 'July'").D4.mean()-df2.query("month == 'June'").D4.mean())/df2.query("month == 'June'").D4.mean()

0.18062573368039606

# Texas 2022 drought observations

* Rate of growth for the state under any drought category (represented by D0) was 3.3% in a 7-month period
* Rate of growth for the state under exceptional drought category (D4) was 29.6% in a 4-month period or 241.2% in a 5-month period beginning in March when D4 category first began in 2022.
* Biggest jump was March to April with 163.3%
* Then it was 56.8% April to May, -30% to June, and 18.1% to July

In [122]:
df

MapDate StateAbbreviation   None      D0      D1     D2     D3     D4  \
0   20110719                TX   0.00  100.00   98.96  95.78  91.19  75.16   
1   20110712                TX   0.00  100.00   97.43  95.78  90.97  71.66   
2   20110705                TX   2.41   97.59   95.73  94.39  90.21  71.30   
3   20110628                TX   2.68   97.32   95.71  94.52  90.62  72.32   
4   20110621                TX   3.33   96.67   95.71  94.52  91.31  70.61   
5   20110614                TX   1.98   98.02   96.52  94.76  88.56  64.78   
6   20110607                TX   1.98   98.02   96.52  94.04  85.41  57.83   
7   20110531                TX   2.25   97.75   96.07  91.89  81.09  50.65   
8   20110524                TX   2.30   97.70   95.77  91.97  78.48  43.97   
9   20110517                TX   0.00  100.00   97.01  92.40  80.01  47.87   
10  20110510                TX   0.00  100.00   97.78  93.88  82.06  47.55   
11  20110503                TX   0.00  100.00   98.86  93.99  73.74  25.96   
12  20110426                TX   0.00  100.00  100.00  94.97  70.42  17.16   
13  20110419                TX   0.00  100.00  100.00  91.52  67.97  14.99   
14  20110412                TX   0.00  100.00   97.94  86.44  60.58  10.03   
15  20110405                TX   0.00  100.00   97.99  86.50  60.15   4.82   
16  20110329                TX   0.00  100.00   94.87  78.54  43.07   0.00   
17  20110322                TX   1.70   98.30   92.05  64.06  28.98   0.00   
18  20110315                TX   3.23   96.77   83.75  56.99  17.13   0.00   
19  20110308                TX   8.31   91.69   77.99  53.64  14.16   0.00   
20  20110301                TX   4.99   95.01   72.91  46.35  12.72   0.00   
21  20110222                TX   5.08   94.92   69.75  45.25  12.64   0.00   
22  20110215                TX  12.52   87.48   58.12  37.35   7.78   0.00   
23  20110208                TX  15.72   84.28   57.15  37.35   7.78   0.00   
24  20110201                TX  19.47   80.53   53.00  35.11   6.93   0.00   
25  20110125                TX  19.47   80.53   53.00  31.56   8.66   0.00   
26  20110118                TX  23.28   76.72   51.42  25.62   7.30   0.00   
27  20110111                TX  20.16   79.84   59.13  25.37   9.48   0.00   
28  20110104                TX  13.54   86.46   66.68  36.30  13.04   0.00   
29  20101228                TX   7.89   92.11   69.44  37.46   9.59   0.00   

    ValidStart    ValidEnd     month  
0   2011-07-19  2011-07-25      July  
1   2011-07-12  2011-07-18      July  
2   2011-07-05  2011-07-11      July  
3   2011-06-28  2011-07-04      June  
4   2011-06-21  2011-06-27      June  
5   2011-06-14  2011-06-20      June  
6   2011-06-07  2011-06-13      June  
7   2011-05-31  2011-06-06       May  
8   2011-05-24  2011-05-30       May  
9   2011-05-17  2011-05-23       May  
10  2011-05-10  2011-05-16       May  
11  2011-05-03  2011-05-09       May  
12  2011-04-26  2011-05-02     April  
13  2011-04-19  2011-04-25     April  
14  2011-04-12  2011-04-18     April  
15  2011-04-05  2011-04-11     April  
16  2011-03-29  2011-04-04     March  
17  2011-03-22  2011-03-28     March  
18  2011-03-15  2011-03-21     March  
19  2011-03-08  2011-03-14     March  
20  2011-03-01  2011-03-07     March  
21  2011-02-22  2011-02-28  February  
22  2011-02-15  2011-02-21  February  
23  2011-02-08  2011-02-14  February  
24  2011-02-01  2011-02-07  February  
25  2011-01-25  2011-01-31   January  
26  2011-01-18  2011-01-24   January  
27  2011-01-11  2011-01-17   January  
28  2011-01-04  2011-01-10   January  
29  2010-12-28  2011-01-03  December

In [123]:
df2011 = df.drop(['MapDate', 'StateAbbreviation' , 'None', 'ValidStart','ValidEnd'], axis=1)
df2011

D0      D1     D2     D3     D4     month
0   100.00   98.96  95.78  91.19  75.16      July
1   100.00   97.43  95.78  90.97  71.66      July
2    97.59   95.73  94.39  90.21  71.30      July
3    97.32   95.71  94.52  90.62  72.32      June
4    96.67   95.71  94.52  91.31  70.61      June
5    98.02   96.52  94.76  88.56  64.78      June
6    98.02   96.52  94.04  85.41  57.83      June
7    97.75   96.07  91.89  81.09  50.65       May
8    97.70   95.77  91.97  78.48  43.97       May
9   100.00   97.01  92.40  80.01  47.87       May
10  100.00   97.78  93.88  82.06  47.55       May
11  100.00   98.86  93.99  73.74  25.96       May
12  100.00  100.00  94.97  70.42  17.16     April
13  100.00  100.00  91.52  67.97  14.99     April
14  100.00   97.94  86.44  60.58  10.03     April
15  100.00   97.99  86.50  60.15   4.82     April
16  100.00   94.87  78.54  43.07   0.00     March
17   98.30   92.05  64.06  28.98   0.00     March
18   96.77   83.75  56.99  17.13   0.00     March
19   91.69   77.99  53.64  14.16   0.00     March
20   95.01   72.91  46.35  12.72   0.00     March
21   94.92   69.75  45.25  12.64   0.00  February
22   87.48   58.12  37.35   7.78   0.00  February
23   84.28   57.15  37.35   7.78   0.00  February
24   80.53   53.00  35.11   6.93   0.00  February
25   80.53   53.00  31.56   8.66   0.00   January
26   76.72   51.42  25.62   7.30   0.00   January
27   79.84   59.13  25.37   9.48   0.00   January
28   86.46   66.68  36.30  13.04   0.00   January
29   92.11   69.44  37.46   9.59   0.00  December

In [125]:
df2011.to_csv('texas-2011-dm-data.csv')

In [124]:
df2022 = df2.drop(['MapDate', 'StateAbbreviation' , 'None', 'ValidStart','ValidEnd'], axis=1)
df2022

D0     D1     D2     D3     D4     month
0   99.18  94.77  84.08  56.67  20.40      July
1   99.24  93.82  75.70  51.80  21.32      July
2   97.53  86.79  66.05  45.91  16.11      July
3   96.29  86.39  64.99  43.79  15.82      June
4   92.99  81.18  64.62  43.83  17.11      June
5   91.41  79.97  64.03  42.47  16.78      June
6   88.25  78.81  64.99  40.11  15.60      June
7   85.89  78.44  66.35  44.07  17.91       May
8   87.08  79.11  66.94  46.05  22.08       May
9   90.43  81.42  69.47  55.79  28.91       May
10  89.46  79.23  68.09  52.96  24.53       May
11  91.17  80.02  67.29  50.91  23.19       May
12  93.79  82.11  70.28  50.25  19.62     April
13  96.13  87.38  72.46  53.54  15.91     April
14  97.13  87.66  74.12  49.11  14.20     April
15  95.05  84.73  71.45  40.56   9.78     April
16  95.10  88.22  70.79  42.10   7.03     March
17  95.72  88.24  71.31  40.38   6.20     March
18  96.31  91.22  70.13  41.20   8.63     March
19  96.05  89.93  68.43  36.38   6.39     March
20  93.34  80.71  56.71  24.47   0.00     March
21  92.33  79.06  52.89  22.44   0.00  February
22  88.17  77.61  55.58  24.53   0.00  February
23  88.17  78.09  55.02  23.88   0.00  February
24  92.96  83.79  69.20  31.56   0.00  February
25  96.29  86.71  67.46  31.84   0.00   January
26  96.47  86.04  64.67  25.87   0.00   January
27  96.79  82.48  62.44  21.91   0.00   January
28  92.42  79.83  54.25  16.69   0.00   January
29  86.98  67.27  36.58  10.65   0.00  December

In [127]:
df2022.to_csv('texas-2022-dm-data.csv')

# After reviewing the pct_change data in DataWrapper ...
* I initially thought the best way to see how we could show if 2022 is in the path to become anything like 2011, even if later in the year, was to go through rate of change.
* After seeing the data in various viz ... it might not be it.
* So now, changing gears to see month by month averages for D0 and D4.

In [188]:
# Let's turn our averages into df format
df22_d4 = df2.groupby('month').D4.mean().sort_index().to_frame().reset_index()
df11_d4 = df.groupby('month').D4.mean().sort_index().to_frame().reset_index()

AttributeError: 'DataFrameGroupBy' object has no attribute 'D4'

In [193]:
# Ah we may have accidentally saved name change to original dataframes, so going to change that now
df.rename(columns={'D411_pctchng':'D4'}, inplace=True)
df2.rename(columns={'D422_pctchng':'D4'}, inplace=True)

In [196]:
# Let's try again ... to turn our averages into df format
df22_d4 = df2.groupby('month').D4.mean().sort_index().to_frame().reset_index()
df11_d4 = df.groupby('month').D4.mean().sort_index().to_frame().reset_index()

In [197]:
# Let's rename columns of new dataframes so when we merge, we know the difference
df11_d4.rename(columns={'D4':'D4_2011'}, inplace=True)
df22_d4.rename(columns={'D4':'D4_2022'}, inplace=True)

In [199]:
# Let's merge the new dataframes 
merged_d4 = df11_d4.merge(df22_d4, on='month')
merged_d4

month    D4_2011    D4_2022
0    01   0.000000   0.000000
1    02   0.000000   0.000000
2    03   0.000000   5.650000
3    04  11.750000  14.877500
4    05  43.200000  23.324000
5    06  66.385000  16.327500
6    07  72.706667  19.276667
7    12   0.000000   0.000000

In [200]:
merged_d4.to_csv('texas-d4-11-22.csv')

In [202]:
# Now we'll do it for D0 too, just to add some dynamic data and to see if the drought in general is comparable...
df22_d0 = df2.groupby('month').D0.mean().sort_index().to_frame().reset_index()
df11_d0 = df.groupby('month').D0.mean().sort_index().to_frame().reset_index()
df11_d0.rename(columns={'D0':'D0_2011'}, inplace=True)
df22_d0.rename(columns={'D0':'D0_2022'}, inplace=True)
merged_d0 = df11_d0.merge(df22_d0, on='month')
merged_d0

month     D0_2011  D0_2022
0    01   80.887500  95.4925
1    02   86.802500  90.4075
2    03   96.354000  95.3040
3    04  100.000000  95.5250
4    05   99.090000  88.8060
5    06   97.507500  92.2350
6    07   99.196667  98.6500
7    12   92.110000  86.9800

In [203]:
merged_d0.to_csv('texas-d0-11-22.csv')

In [8]:
# Now we'll do it for D1 too, just to add some dynamic data and to see if the drought in general is comparable...
df11_d1 = df.groupby('month').D1.mean().sort_index().to_frame().reset_index()
df22_d1 = df2.groupby('month').D1.mean().sort_index().to_frame().reset_index()
df11_d1.rename(columns={'D1':'D1_2011'}, inplace=True)
df22_d1.rename(columns={'D1':'D1_2022'}, inplace=True)
merged_d1 = df11_d1.merge(df22_d1, on='month')
merged_d1

month    D1_2011    D1_2022
0    01  57.557500  83.765000
1    02  59.505000  79.637500
2    03  84.314000  87.664000
3    04  98.982500  85.470000
4    05  97.098000  79.644000
5    06  96.115000  81.587500
6    07  97.373333  91.793333
7    12  69.440000  67.270000

In [9]:
merged_d1.to_csv('texas-d1-11-22.csv')